ckanext/xloader/plugin.py

```diff
@@ -212,3 +229,40 @@ def get_helpers(self):
             "xloader_status": xloader_helpers.xloader_status,
             "xloader_status_description": xloader_helpers.xloader_status_description,
         }
+
+
+def _should_remove_unsupported_resource_from_datastore(res_dict_or_obj):
+    if not toolkit.asbool(toolkit.config.get('ckanext.xloader.clean_datastore_tables', False)):
+        return False
+    if isinstance(res_dict_or_obj, Resource):
+        res_dict_or_obj = res_dict_or_obj.__dict__
+    return ((not XLoaderFormats.is_it_an_xloader_format(res_dict_or_obj.get('format', u''))
+                or res_dict_or_obj.get('url_changed', False))
+            and (res_dict_or_obj.get('url_type') == 'upload'
+                or res_dict_or_obj.get('url_type') == '')
+            and (res_dict_or_obj.get('datastore_active', False)
+                or res_dict_or_obj.get('extras', {}).get('datastore_active', False)))
+
+
+def _remove_unsupported_resource_from_datastore(resource_id):
+    """
+    Callback to remove unsupported datastore tables.
+    Controlled by config value: ckanext.xloader.clean_datastore_tables.
+    Double check the resource format. Only supported Xloader formats should have datastore tables.
+    If the resource format is not supported, we should delete the datastore tables.
+    """
+    lc = ckanapi.LocalCKAN()
+    try:
+        res = lc.action.resource_show(id=resource_id)
+    except toolkit.ObjectNotFound:
+        log.error('Resource %s does not exist.' % res['id'])
+        return
+
+    if _should_remove_unsupported_resource_from_datastore(res):
+        log.info('Unsupported resource format "{}". Deleting datastore tables for resource {}'
```

Comment:
This should use parameterised logging with `%s` instead of `format`.

Reply:
Done